src/webgpu/web_platform/external_texture/video.spec.ts

```diff
@@ -3,13 +3,241 @@ Tests for external textures from HTMLVideoElement (and other video-type sources?
+      t.device.queue.submit([commandEncoder.finish()]);
+
+      // Top left corner should be red. Sample a few pixels away from the edges to avoid color
+      // management artifacts.
+      t.expectSinglePixelIn2DTexture(
+        colorAttachment,
+        kFormat,
+        { x: 2, y: 2 },
+        {
+          exp: new Uint8Array([0xff, 0x00, 0x00, 0xff]),
+        }
+      );
+
+      // Bottom right corner should be green. Sample a few pixels away from the edges to avoid color
+      // management artifacts.
+      t.expectSinglePixelIn2DTexture(
+        colorAttachment,
+        kFormat,
+        { x: kWidth - 3, y: kHeight - 3 },
+        {
+          exp: new Uint8Array([0x00, 0xff, 0x00, 0xff]),
+        }
+      );
+
+      // This function submits the same render pass as above, but should result in an error because the
+      // GPUExternalTexture will be destroyed via microtask upon requestAnimationFrame.
+      const ensureDestroyed = function () {
+        const commandEncoder2 = t.device.createCommandEncoder();
+        const passEncoder2 = commandEncoder2.beginRenderPass({
+          colorAttachments: [
+            {
+              view: colorAttachment.createView(),
+              loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 1.0 },
+              storeOp: 'store',
+            },
+          ],
+        });
+        passEncoder2.setPipeline(pipeline);
+        passEncoder2.setBindGroup(0, bindGroup);
+        passEncoder2.draw(6);
+        passEncoder2.endPass();
+
+        t.expectGPUError('validation', () => t.device.queue.submit([commandEncoder2.finish()]));
+      };
+
+      requestAnimationFrame(ensureDestroyed);
+    });
+  });
+
+g.test('importExternalTexture,compute')
+  .desc(
+    `
+Tests that we can import an HTMLVideoElement into a GPUExternalTexture and use it in a compute shader.
+`
+  )
+  .fn(async t => {
+    const videoUrl = getResourcePath('red-green.webmvp8.webm');
+    const video = document.createElement('video');
+    video.src = videoUrl;
+
+    await startPlayingAndWaitForVideo(video, () => {
+      const externalTextureDescriptor = { source: video };
+      const externalTexture = t.device.importExternalTexture(externalTextureDescriptor);
+
+      const result = t.device.createBuffer({
+        size: 8,
+        usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.STORAGE,
+      });
+
+      const pipeline = t.device.createComputePipeline({
+        compute: {
+          module: t.device.createShaderModule({
+            code: `
+              [[block]] struct Data {
+                  height : i32;
+                  width : i32;
+              };
+    
+              [[group(0), binding(0)]] var t : texture_external;
+              [[group(0), binding(1)]] var<storage, read_write> dst : Data;
+    
+              [[stage(compute), workgroup_size(1)]] fn main() {
+                var dims : vec2<i32> = textureDimensions(t);
+                dst.width = dims.x;
+                dst.height = dims.y;
```

Comment:
Ideally copy the data into a storage texture and check the contents? That way we can be sure the sampling is working in compute shaders when we implement the 0-copy impl (which might differ slightly for compute shaders since iirc texture sampling of the underlying textures won't be possible in compute).

If not implementing this now, please leave a TODO before landing the PR.

Reply:
Added TODO